fedbiomed/node/dataset_manager/_db_dataclasses.py

```
@@ -0,0 +1,107 @@
+# This file is originally part of Fed-BioMed
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Dataclasses for database entities
+"""
+
+import uuid
+from dataclasses import asdict, dataclass, fields
+from typing import Any, Dict, List, Optional, Union, get_origin
+
+from fedbiomed.common.constants import ErrorNumbers
+from fedbiomed.common.exceptions import FedbiomedError
+
+
+class TableEntry:
+    """Base class for database entries with common conversion methods"""
+
+    def to_dict(self) -> dict:
+        """Convert entry to dictionary - removes None values"""
+        return {k: v for k, v in asdict(self).items() if v is not None}
+
+    @classmethod
+    def validate_required_fields(cls, data: dict) -> None:
+        """Validate that all required fields are present"""
+        # Auto-extract required fields from dataclass definition
+        required_fields = [
+            field.name
+            for field in fields(cls)
+            if (
+                field.default == field.default_factory
+                and get_origin(field.type) not in (Union, type(Union[str, None]))
+            )
+        ]
+
+        missing_fields = [field for field in required_fields if field not in data]
+        if missing_fields:
+            raise FedbiomedError(
+                f"{ErrorNumbers.FB632.value}: Validation failed for {cls.__name__}: "
+                f"Missing required fields: {missing_fields}"
+            )
```

Hi @leonard2103,

Do you need this method. For example, executing following DatasetEntry raises error due to missing fields.

```
DatasetEntry(name="hello", data_type="csv", tags=["x","y"])
```

---

The raise will be raise based on which values from the dataclass (DatasetEntry, ...) are left optional. Just required parameters should raise the Error. The dataclasses can be modified based on the values that should always be present.

---

Hi @leonard2103, 
I didn't understand the intention.  

```
@dataclass
class MyDataClass: 
   value_1: str
   test: Optiona[str] = None
```

The following execution will not raise error

```
MyDataClass(value_1 = "my-value")
``` 

you can even do 

```
class MyDataClass: 
   value_1: str
   test: str = 'my-default-value'

MyDataClass(value_1 = "my-value")
```

but you can not do:

```
MyDataClass(test= "my-value")
```
because `value_1` is required. 


And `MyDataClass` will have the value of `test` as `None` or any value that is assigned as default.